kartothek/core/index.py

```diff
@@ -132,15 +132,13 @@ def __repr__(self) -> str:
             class_=type(self).__name__, attrs=", ".join(repr_str)
         )
 
-    def observed_values(self) -> np.array:
+    def observed_values(self, date_as_object=True) -> np.array:
         """
         Return an array of all observed values
         """
-        return np.fromiter(
-            (self.normalize_value(self.dtype, x) for x in self.index_dct.keys()),
-            count=len(self.index_dct),
-            dtype=self.dtype.to_pandas_dtype(),
-        )
+        keys = np.array(list(self.index_dct.keys()))
```

Comment:
we can't use `np.fromiter(self.index_dct.keys(), dtype=...)` here?

Reply:
I haven't dived deep into this, but it seems odd that we convert to numpy.array, to pyarrow and to numpy again

Reply:
We can't use the `fromiter` because it cannot allocate the array for object types properly. Didn't dive into it any further. Yes, it is a bit strange and not very performant but I chose the practical fix here over the fast one, especially since we know that this code is type stable

Reply:
I'm okay with merging this now but we should look at the performance of this
```suggestion
        # TODO: Benchmark performance
        keys = np.array(list(self.index_dct.keys()))
```

Reply:
I added an asv benchmark but since there is no baseline, yet, I don't intend to look into it right now. Improvements are welcome, of course